code/src/containers/tables/orders-table/components/orders-table-head/OrdersTableHead.test.tsx

```diff
@@ -86,16 +70,97 @@ describe("OrderTableHead", () => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[1][0]);
+      expect(params.get("sort")).toBe("createdAt,desc");
+    });
+
+    fireEvent.click(sortButtonTotalPrice);
+    await waitFor(() => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[2][0]);
+      expect(params.get("sort")).toBe("total,asc");
+    });
+
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      params = new URLSearchParams(mockSetSearchParams.mock.calls[3][0]);
+      expect(params.get("sort")).toBe("createdAt,asc");
+    });
+  });
+
+  test("initial isActive is set correctly based on URL-sort parameter and icon class updates", () => {
+    renderComponent([
+      "ordersTable.columns.createdAt",
+      "ordersTable.columns.totalPrice"
+    ]);
+    const sortButtonCreatedAt = screen.getAllByRole("button")[0];
+    expect(sortButtonCreatedAt).toHaveClass("Mui-active");
+
+    const sortButtonTotal = screen.getAllByRole("button")[1];
+    expect(sortButtonTotal).not.toHaveClass("Mui-active");
+
+    fireEvent.click(sortButtonTotal);
+
+    waitFor(() => {
+      expect(sortButtonCreatedAt).toHaveClass("Mui-active");
+      expect(sortButtonTotal).not.toHaveClass("Mui-active");
+    });
+    waitFor(() => {
+      fireEvent.click(sortButtonCreatedAt);
+    });
+    expect(sortButtonCreatedAt).toHaveClass("Mui-active");
+    expect(sortButtonTotal).not.toHaveClass("Mui-active");
+  });
+
+  test("sets default sort direction on first render", async () => {
+    (useSearchParams as jest.Mock).mockReturnValue([
+      new URLSearchParams(),
+      mockSetSearchParams
+    ]);
+    renderComponent(sortableFields);
+    await waitFor(() => {
+      expect(mockSetSearchParams).toHaveBeenCalledWith(
+        new URLSearchParams({ sort: "createdAt,desc" })
+      );
+    });
+  });
+
+  test("correctly updates state on sort direction change", async () => {
+    renderComponent(["ordersTable.columns.createdAt"]);
+    const sortButtonCreatedAt = screen.getByText(
+      /ordersTable.columns.createdAt/i
+    );
 
-    expect(sortButton).toHaveClass("MuiTableSortLabel-iconDirectionDesc");
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      expect(mockSetSearchParams).toHaveBeenCalledWith(
+        new URLSearchParams({ sort: "createdAt,asc" })
+      );
+    });
 
-    fireEvent.click(sortButton);
-    expect(sortButton).toHaveClass("MuiTableSortLabel-iconDirectionAsc");
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      expect(mockSetSearchParams).toHaveBeenCalledWith(
+        new URLSearchParams({ sort: "createdAt,desc" })
+      );
+    });
 
-    fireEvent.click(sortButton);
-    expect(sortButton).toHaveClass("MuiTableSortLabel-iconDirectionDesc");
+    fireEvent.click(sortButtonCreatedAt);
+    await waitFor(() => {
+      expect(mockSetSearchParams).toHaveBeenCalledWith(
+        new URLSearchParams({ sort: "createdAt,asc" })
+      );
+    });
```

Comment:
Create some helper for it, it is repeated

Reply:
Created helper function too) Thank you for advice